doc/direction.md

```diff
@@ -16,35 +16,35 @@ These are the current medium-term goals for Luanti development, in no
 particular order.
 
 These goals were created from the top points in a
-[roadmap brainstorm](https://github.com/luanti-org/luanti/issues/10461).
-This should be reviewed approximately yearly, or when goals are achieved.
+[roadmap brainstorm](https://github.com/luanti-org/luanti/issues/16162).
+This is reviewed approximately every two years.
 
 Pull requests that address one of these goals will be labeled as "Roadmap".
 PRs that are not on the roadmap will be closed unless they receive a concept
-approval within a month, issues can be used for preapproval.
-Bug fixes are exempt for this, and are always accepted and prioritized.
+approval within a month. Issues can be used for preapproval.
+Bug fixes are exempt from this, and are always accepted and prioritized.
 See [CONTRIBUTING.md](../.github/CONTRIBUTING.md) for more info.
 
-### 2.1 Rendering/Graphics improvements
+### 2.1 SSCSM
 
-The priority is fixing the issues, performance, and general correctness.
-Once that is done, fancier features can be worked on, such as water shaders,
-shadows, and improved lighting.
+Server-Sent Client-side Modding has been a long requested feature, as it
+allows game developers to highly expand the tools for their creations.
+Implementing SSCSM also means to feature more dehardcoding, furthermore
+allowing us to improve the overall performances of the engine.
 
-Examples include
-[transparency sorting](https://github.com/luanti-org/luanti/issues/95),
-[particle performance](https://github.com/luanti-org/luanti/issues/1414),
-[general view distance](https://github.com/luanti-org/luanti/issues/7222).
+Instead of focusing on short-term solutions that will inevitably lead to more
+technical debt to deal with, SSCSM paves the way for a cleaner architecture
+designed to stay.
 
-This includes work on maintaining
-[our Irrlicht fork](https://github.com/luanti-org/luanti/tree/master/irr), and switching to
-alternative libraries to replace Irrlicht functionality as needed
+### 2.2 Input Handling
 
-### 2.2 Internal code refactoring
+Luanti keys are currently limited to a small subset, not allowing game developers
+to map the majority of the keys a device usually offers. This limits the possibilities
+of game creators, forcing them to either implement a workaround or, worse, forget
+about it.
 
-To ensure sustainable development, Luanti's code needs to be
-[refactored and improved](https://github.com/luanti-org/luanti/pulls?q=is%3Aopen+sort%3Aupdated-desc+label%3A%22Code+quality%22).
-This will remove code rot and allow for more efficient development.
+Using a gamepad also represents a known issue in Luanti, as some devices might not
+work at all or result in an uncomfortable user experience.
```

Comment:
how does this interact with "Instead of focusing on short-term solutions that will inevitably lead to more technical debt to deal with, SSCSM paves the way for a cleaner architecture designed to stay."? 
there's few we can do for input handling that isn't "make SSCSM a thing and add proper input APIs".
or do you have something in mind?

Reply:
It's two different point, they're not related

Reply:
IMO this section can be largely broken down into two parts:
* Improve input handling within the engine
  * Support for (in general) analog input values
  * Support for other modes of input (e.g. joysticks)
* APIs for SSMs/(SS)CSMs, e.g. #12488

Reply:
Improvements on the keybinding system (e.g. support for gamepads and analog input) are not related to SSCSMs; there is not much I can think of otherwise that is unrelated to SSCSMs.

That said, changes that are not directly related to SSCSMs can likely gain approval in other ways (code quality/refactoring; feature with concept approval/supported by coredev; dependency), so I would be fine with not having input handling as a dedicated point on the roadmap.